public/pages/workflows/workflows.tsx

```diff
@@ -54,6 +76,13 @@ function replaceActiveTab(activeTab: string, props: WorkflowsProps) {
  */
 export function Workflows(props: WorkflowsProps) {
   const dispatch = useAppDispatch();
+  const location = useLocation();
+  const queryParams = getDataSourceFromURL(location);
+  const dataSourceEnabled = getDataSourceEnabled().enabled;
+  const [MDSOverviewState, setMDSOverviewState] = useState<MDSStates>({
+    queryParams,
+    selectedDataSourceId: queryParams.dataSourceId,
+  });
```

Comment:
I think this can be simplified a bunch of ways. `MDSStates` is redundantly capturing the data source ID, both as a nested field under `queryParams.dataSourceId` and in the standalone `selectedDataSourceId` field. Only `selectedDataSourceId` is ever used. This could instead just persist the data source ID string:
```
const [dataSourceId, setDataSourceId] = useState<string>(queryParams.dataSourceId);
```

All of the references to it can be changed from `MDSOverviewState.selectedDataSourceId` -> `dataSourceId`.

The one place where this state is updated can just be changed to
```
setDataSourceId(dataSourceId)
```

Reply:
And I think that means `MDSStates` can be removed altogether; I don't think it's used anywhere else but you can double check.